static_src/components/form.jsx

```diff
@@ -0,0 +1,257 @@
+
```

Comment:
This whole thought came from me thinking about how to test this. It's generic enough for others to use it. But the tests for this would stick out from the tests for the rest of the code. (But I'm not sure if it's too opinionated).
Ok I'm done with this rant lol.